ObjectiveGit/GTNote.m

```diff
@@ -0,0 +1,113 @@
+//
+//  GTNote.m
+//  ObjectiveGitFramework
+//
+//  Created by Slava Karpenko on 16.05.16.
+//  Copyright © 2016 Wildbit LLC. All rights reserved.
+//
+
+#import "GTNote.h"
+#import "NSError+Git.h"
+#import "GTSignature.h"
+#import "GTReference.h"
+#import "GTRepository.h"
+#import "NSString+Git.h"
+#import "GTOID.h"
+
+#import "git2/errors.h"
+#import "git2/notes.h"
+
+@interface GTNote ()
+{
+	git_note *_note;
+}
+
+@end
+@implementation GTNote
+
+- (NSString *)description {
+	return [NSString stringWithFormat:@"<%@: %p>", NSStringFromClass([self class]), self];
+}
+
+#pragma mark API
+
+- (void)dealloc {
+	if (_note != NULL) {
+		git_note_free(_note);
+	}
+}
+
+- (git_note *)git_note {
+	return _note;
+}
+
+- (NSString*)note {
+	return @(git_note_message(self.git_note));
+}
+
+- (GTSignature *)author {
+	return [[GTSignature alloc] initWithGitSignature:git_note_author(self.git_note)];
+}
+
+- (GTSignature *)committer {
+	return [[GTSignature alloc] initWithGitSignature:git_note_committer(self.git_note)];
+}
+
+- (GTOID*)targetOID {
+	return [GTOID oidWithGitOid:git_note_id(self.git_note)];
+}
+
+- (instancetype)initWithTargetOID:(GTOID *)oid repository:(GTRepository *)repository referenceName:(NSString *)referenceName error:(NSError **)error {
+	int err = GIT_OK;
+	
+	id object = [self initWithTargetGitOID:(git_oid *)oid.git_oid repository:repository.git_repository referenceName:referenceName.UTF8String error:&err];
+	
+	if (err != GIT_OK && error != NULL) {
+		*error = [NSError git_errorFor:err description:@"Failed to create a note."];
+	}
+	
+	return object;
+}
+
+- (instancetype)initWithTargetGitOID:(git_oid *)oid repository:(git_repository *)repository referenceName:(const char *)referenceName error:(int *)error {
+	self = [super init];
+	if (self == nil) return nil;
+	
+	int gitErr = git_note_read(&_note, repository, referenceName, oid);
+	
+	if (gitErr != GIT_OK) {
+		if (error != NULL) *error = gitErr;
+		
+		return nil;
+	}
+	
+	return self;
+}
+
+- (instancetype)init {
+	NSAssert(NO, @"Call to an unavailable initializer.");
+	return nil;
+}
+
++ (NSString*)defaultReferenceNameForRepository:(GTRepository *)repository error:(NSError **)error {
+	NSString* noteRef = nil;
+	
+	git_buf default_ref_name = { 0 };
+	int gitErr = git_note_default_ref(&default_ref_name, repository.git_repository);
+	if (gitErr != GIT_OK) {
+		git_buf_free(&default_ref_name);
```

Comment:
Who knows? After all, if it failed filling a pointer, `git_buf_free` would be a no-op, so it may be redundant, but definitely there's no harm in doing that. Besides, it's a pattern I picked from other parts of the code (see, for example, GTRepository.m:604 in `- (NSString *)preparedMessageWithError:(NSError **)error`.